crates/polars-core/src/utils/mod.rs

```diff
@@ -141,7 +141,8 @@ impl Container for DataFrame {
     }
 
     fn chunk_lengths(&self) -> impl Iterator<Item = usize> {
-        self.get_columns()[0].chunk_lengths()
+        // @scalar-correctness?
+        self.columns[0].as_materialized_series().chunk_lengths()
```

Comment:
Seems wasteful to materialize here. I think we should create e special `chunk_lengths` iterator for `Column`. For `Column::Scalar` it returns 1 time `len`.